src/Features/Core/Portable/SemanticSearch/Tools/ReferencingSyntaxFinder.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Threading;
+
+namespace Microsoft.CodeAnalysis.SemanticSearch;
+
+internal sealed class ReferencingSyntaxFinder(Solution solution, CancellationToken cancellationToken)
+{
+    private static readonly FindReferencesSearchOptions s_options = new()
+    {
+        AssociatePropertyReferencesWithSpecificAccessor = false,
+        Cascade = false,
+        DisplayAllDefinitions = false,
+        Explicit = true,
+        UnidirectionalHierarchyCascade = false
+    };
+
+    public IEnumerable<SyntaxNode> Find(ISymbol symbol)
+        => FindAsync(symbol).ToBlockingEnumerable(cancellationToken);
+
+    public async IAsyncEnumerable<SyntaxNode> FindAsync(ISymbol symbol)
```

Comment:
i believ ethis can be simplified.  I"ve opened https://github.com/dotnet/roslyn/pull/78060/commits/7d79ff4a3df9c3467d3782c2050f269e200fbefd to demonstrate.  but can do as followup.